### PR TITLE
Use solana wallet adapter methods instead of window.solana

### PR DESCRIPTION
**Pull request by ArpitAgrawal321**
This PR:-

- Removes all instances of window.solana and replaces them with methods provided by solana wallet adapter.
- Updated various methods to use wallet adapter.

### CODE REVIEW

**Comment by kespinola** [comment]
@ArpitAgrawal321 thanks so much for this. I'm going to get it across the line for you.

**Comment by metallicalfa2** [comment]
@kespinola  Further comments:
- Currently, in the dev setup, the network for the adapters is hardcoded to mainnet. (in _app.tsx)
- All the adapter method return a Uint8Array. We should update the Signers to use the same instead of Buffer. 
- Support >= node@16 instead of specific node version (is there a reason for this ?)

File: pages/storefront/edit.tsx [hunk 1]
```diff
@@ -82,13 +84,13 @@ export default function Edit() {
...
 
-  if (isNil(solana) || isNil(storefront) || isNil(wallet)) {
+  if (isNil(storefront) || isNil(wallet)) {
     return (
       <Row justify="center">
         <Card>
           <Space direction="vertical">
             <Paragraph>Connect your Solana wallet to edit your store.</Paragraph>
-            <Button type="primary" block onClick={connect}>
+            <Button type="primary" block onClick={()=>connect()}>
```
Comment (kevinrodriguez-io) [inline pages/storefront/edit.tsx:93]:
```suggestion
            <Button type="primary" block onClick={() => connect()}>
```

File: pages/storefront/edit.tsx [hunk 2]
```diff
@@ -99,7 +101,7 @@ export default function Edit() {
 
   const values = reduceFieldData(fields);
 
-  const subdomainUniqueness = validateSubdomainUniqueness(ar, wallet.pubkey);
+  const subdomainUniqueness = validateSubdomainUniqueness(ar, publicKey?.toString());
```
Comment (kevinrodriguez-io) [inline pages/storefront/edit.tsx:104]: We probably want to check there's a pubKey before reaching this point, but I'm not pretty sure

File: pages/storefront/new.tsx [hunk 3]
```diff
@@ -67,13 +70,13 @@ export default function New() {
...
 
-  if (isNil(solana) || isNil(wallet)) {
+  if (isNil(wallet) || !connected) {
     return (
       <Row justify="center">
         <Card>
           <Space direction="vertical">
             <Paragraph>Connect your Solana wallet to create a store.</Paragraph>
-            <Button type="primary" block onClick={connect}>
+            <Button type="primary" block onClick={()=>connect()}>
```
Comment (kevinrodriguez-io) [inline pages/storefront/new.tsx:79]:
```suggestion
            <Button type="primary" block onClick={() => connect()}>
```

File: src/common/components/elements/Upload.tsx [hunk 4]
```diff
@@ -14,7 +14,7 @@ type UploadProps = {
 };
 
 export default function FileUpload({ children, value, onChange }: UploadProps) {
-  const { solana } = useContext(WalletContext);
+  const contextState = useWallet();
```
Comment (kevinrodriguez-io) [inline src/common/components/elements/Upload.tsx:17]: Weird name, shouldn't it just be wallet?

File: src/modules/arweave/upload.ts [hunk 5]
```diff
@@ -52,10 +52,10 @@ export const uploadFile = async ({
 
     const payload: ArweaveUploadPayload = {
       fileHash: hash.toString('base64'),
-      pubkey: solana.publicKey.toBase58(),
+      pubkey: wallet.publicKey?.toString() || "", 
     };
 
-    const notarized: ArweaveUploadParams = await notarize(payload, signPhantom(solana), {
+    const notarized: ArweaveUploadParams = await notarize(payload, wallet.signMessage as Signer, {
```
Comment (kevinrodriguez-io) [inline src/modules/arweave/upload.ts:58]: This really works?

File: src/modules/storefront/provider.tsx [hunk 6]
```diff
@@ -3,36 +3,35 @@ import { initArweave } from '@/modules/arweave';
...
   children: (props: StorefrontProviderChildrenProps) => React.ReactElement;
 };
 
-export const StorefrontProvider = ({ wallet, children }: StorefrontProviderProps) => {
+export const StorefrontProvider = ({ children }: StorefrontProviderProps) => {
   const [searching, setSearching] = useState(false);
   const [storefront, setStorefront] = useState<Storefront>();
   const arweave = initArweave();
-  const router = useRouter();
+  const { publicKey} = useWallet();
```
Comment (kevinrodriguez-io) [inline src/modules/storefront/provider.tsx:22]:
```suggestion
  const { publicKey } = useWallet();
```

File: src/modules/storefront/provider.tsx [hunk 7]
```diff
@@ -3,36 +3,35 @@ import { initArweave } from '@/modules/arweave';
...
 
   useEffect(() => {
-    if (!process.browser || !wallet) {
+    const pub_key = publicKey?.toString();
```
Comment (kevinrodriguez-io) [inline src/modules/storefront/provider.tsx:25]:
```suggestion
    const pubKey = publicKey?.toString();
```

File: src/modules/storefront/provider.tsx [hunk 8]
```diff
@@ -3,36 +3,35 @@ import { initArweave } from '@/modules/arweave';
...
+    if (!process.browser || !pub_key) {
       return;
     }
 
     setSearching(true);
 
     arweaveSDK
       .using(arweave)
-      .storefront.find('solana:pubkey', wallet.pubkey)
+      .storefront.find('solana:pubkey', pub_key)
```
Comment (kevinrodriguez-io) [inline src/modules/storefront/provider.tsx:34]:
```suggestion
      .storefront.find('solana:pubkey', pubKey)
```

File: src/modules/wallet/provider.tsx [hunk 9]
```diff
@@ -23,102 +20,67 @@ const upsertWallet = async (pubkey: string) => {
       return walletSDK.create(pubkey);
```
Comment (kevinrodriguez-io) [inline src/modules/wallet/provider.tsx:20]: We probably also want to get rid of this walletSDK thing

File: src/modules/wallet/provider.tsx [hunk 10]
```diff
@@ -23,102 +20,67 @@ const upsertWallet = async (pubkey: string) => {
...
+          if (!pub_key) throw new Error('Wallet not connected');
+          await upsertWallet(pub_key);
+          
+          const sf = await arweaveSDK.using(arweave).storefront.find('solana:pubkey', pub_key);
+          if (sf) setStorefront(sf);
+          setVerifying(false);
+          if (redirect.current) {
+            const path = redirect.current;
+            redirect.current = '';
+            return router.push(path);
```
Comment (kevinrodriguez-io) [inline src/modules/wallet/provider.tsx:51]:
```suggestion
  useEffect(() => {
    (async () => {
      if (connected && publicKey) {
        setVerifying(true);
        try {
          const pubKey = publicKey?.toString();
          if (!pubKey) throw new Error('Wallet not connected');
          await upsertWallet(pubKey);
          
          const sf = await arweaveSDK.using(arweave).storefront.find('solana:pubkey', pubKey);
          if (sf) setStorefront(sf);
          setVerifying(false);
          if (redirect.current) {
            const path = redirect.current;
            redirect.current = '';
            return router.push(path);
```

**Review by kevinrodriguez-io — CHANGES_REQUESTED** [review]
Looks good overall; please address the given comments, I'll run some end to end tests to check everything is working fine.

**Comment by kespinola** [comment]
We got all this changes in on this PR. A few moving pieces right now for us.

https://github.com/holaplex/holaplex/pull/224

**Pull request closed without merging**